corfudb-tools/src/main/java/org/corfudb/compactor/CorfuStoreCompactorMain.java

```diff
@@ -0,0 +1,412 @@
+
+                Set<TableName> allTableNames = registryTable.keySet();
+                allTableNames.forEach(tableName -> {
+                    CorfuRecord<TableDescriptors, TableMetadata> registryRecord = registryTable.get(tableName);
+                    TableDescriptors.Builder tableDescriptorsBuilder = TableDescriptors.newBuilder();
+
+                    registryRecord.getPayload().getFileDescriptorsMap().forEach(
+                        (protoName, fileDescriptorProto) -> {
+                        // populate ProtobufDescriptorTable
+                        ProtobufFileName fileName = ProtobufFileName
+                            .newBuilder().setFileName(protoName).build();
+                        ProtobufFileDescriptor fileDescriptor = ProtobufFileDescriptor
+                            .newBuilder().setFileDescriptor(fileDescriptorProto).build();
+                        CorfuRecord<ProtobufFileDescriptor, TableMetadata> corfuRecord =
+                            descriptorTable.putIfAbsent(fileName, new CorfuRecord<>(fileDescriptor, null));
+                        if (corfuRecord == null) {
+                            log.info("Add proto file {}, fileDescriptor {} to ProtobufDescriptorTable",
+                                fileName, fileDescriptor.getFileDescriptor());
+                        }
+                        // construct a new tableDescriptorsMap using default FileDescriptorProto instances
+                        tableDescriptorsBuilder.putFileDescriptors(protoName,
+                            fileDescriptorProto.getDefaultInstanceForType());
+                    });
+
+                    tableDescriptorsBuilder.setKey(registryRecord.getPayload().getKey());
+                    tableDescriptorsBuilder.setValue(registryRecord.getPayload().getValue());
+                    tableDescriptorsBuilder.setMetadata(registryRecord.getPayload().getMetadata());
+
+                    // clean up FileDescriptorsMap inside RegistryTable to optimize memory
+                    TableDescriptors tableDescriptors = tableDescriptorsBuilder.build();
+                    registryTable.put(tableName, new CorfuRecord<>(tableDescriptors, registryRecord.getMetadata()));
+
+                    log.info("Cleaned up an entry in RegistryTable: {}${}",
+                            tableName.getNamespace(), tableName.getTableName());
+                });
+                tx.commit();
+                log.info("syncProtobufDescriptorTable: completed!");
+                break;
+            } catch (TransactionAbortedException txAbort) {
+                if (numRetries-- <= 0) {
+                    throw txAbort;
+                }
+                log.info("syncProtobufDescriptorTable: commit failed. " +
+                    "Will retry {} times. Cause {}", numRetries, txAbort);
+            }
+        }
+    }
+
+    private void getCompactorArgs(String[] args) {
```

Comment:
Why not use the @Parameter annotation as done in UfoCompactorMain.java?

Reply:
That requires a new import com.beust.jcommander.JCommander and I did not want to add a new package dependency just for parsing command line arguments.